src/agent/config.js

```diff
@@ -0,0 +1,206 @@
+   * @memberof DebugAgentConfig
+   * @default
+   */
+  logLevel: 1,
+
+  /**
+   * @property {number=} How frequently should the list of breakpoints be
+   * refreshed from the cloud debug server.
+   * @memberof DebugAgentConfig
+   * @default
+   */
+  breakpointUpdateIntervalSec: 10,
+
+  /**
+   * @property {number=} breakpoints and logpoints older than this number of
+   * seconds will be expired on the server.
+   * @memberof DebugAgentConfig
+   * @default
+   */
+  breakpointExpirationSec: 60 * 60 * 24,  // 24 hours
+
+  /**
+   * @property {object=} configuration options on what is captured on a
+   * snapshot.
+   * @memberof DebugAgentConfig
+   */
+  capture: {
+    /**
+     * @property {boolean=} Whether to include details about stack frames
+     * belonging to node-core.
+     * @default
+     */
+    includeNodeModules: false,
+
+
+    /**
+     * @property {number=} Maximum number of stack frames to capture data for.
+     * The limit is aimed to reduce overall capture time.
+     * @default
+     */
+    maxFrames: 20,
+
+    /**
+     * @property {number=} We collect locals and arguments on a few top frames.
+     * For the rest only collect the source location
+     * @default
+     */
+    maxExpandFrames: 5,
+
+    /**
+     * @property {number=} To reduce the overall capture time, limit the number
+     * of properties gathered on large objects. A value of 0 disables the limit.
+     * @default
+     */
+    maxProperties: 10,
+
+    /**
+     * @property {number=} Total 'size' of data to gather. This is NOT the
+     * number of bytes of data that are sent over the wire, but instead a very
+     * very coarse approximation based on the length of names and values of the
+     * properties. This should be somewhat proportional to the amount of
+     * processing needed to capture the data and subsequently the network
+     * traffic. A value of 0 disables the limit.
+     * @default
+     */
+    maxDataSize: 20000,
+
+    /**
+     * @property {number=} To limit the size of the buffer, we truncate long
+     * strings. A value of 0 disables truncation.
+     * @default
+     */
+    maxStringLength: 100
+  },
+
+  /**
+   * @property {object=} options affecting log points.
+   * @memberof DebugAgentConfig
+   */
+  log: {
+    /**
+     * @property {number=} The maximum number of logs to record per second per
+     * logpoint.
+     * @memberof DebugAgentConfig
+     * @default
+     */
+    maxLogsPerSecond: 50,
+
+    /**
+     * @property {number=} Number of seconds to wait after the
+     * `maxLogsPerSecond` rate is hit before logging resumes per logpoint.
+     * @default
+     */
+    logDelaySeconds: 1
+  },
+
+  // FIXME(ofrobots): stop accepting this property here
+  // A path to a key file relative to the current working directory. If this
+  // field is set, the contents of the pointed file will be used for
+  // authentication instead of your application default credentials.
+  keyFilename: null,
```

Comment:
Some of these initialize with `null` and others with `undefined`. It would be good to go with one. My preference is `null`.